data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -240,13 +274,22 @@ numerical_equals#endarg
             id={ID1}
         [/filter]
 
-        {ASSERT {VARIABLE_CONDITIONAL {ID1}_{ID2}_strikes {COMPARE} {STRIKES}}}
+        {ASSERT {VARIABLE_CONDITIONAL {ID1}_{ID2}_strikes {COMPARE} $strikes[$i].value}}
         {VARIABLE_OP {ID1}_{ID2}_count add 1}
         [if]
-            {VARIABLE_CONDITIONAL {ID1}_{ID2}_count numerical_equals {WEAPONS}}
+            {VARIABLE_CONDITIONAL {ID1}_{ID2}_count numerical_equals $strikes.length}
             [then]
                 {RESULT}
             [/then]
         [/if]
     [/event]
+
+    # if one of the units dies before all weapons are used, give this result
+    # otherwise there will be an error about [test_do_attack] not matching a unit
```

Comment:
For the two tests that use this, an alternative would be to have Bob attack Alice, as he has only one weapon.

Reply:
That defeats the purpose of testing weapon specials as abilities - the point of that feature is that setting them as an ability affects all weapons.

Reply:
It would defeat the purpose of using an ability instead of a weapon special for normal gameplay, however why does defeat the purpose of using it for testing? Regardless of the number of weapons, we want to know if it goes on until one of them dies.